src/ray/util/pipe_logger.h

```diff
@@ -103,25 +141,26 @@ class RotationFileHandle {
  private:
   MEMFD_TYPE_NON_UNIQUE write_handle_;
```

Comment:
we can use std::optional to indicate invalid write_handle_ so we don't need to write almost duplicate code for linux and windows?

Reply:
updated to INVALID_FD